pennylane/ops/qubit/special_unitary.py

```diff
@@ -0,0 +1,347 @@
+
+        if len(theta_shape) not in {1, 2}:
+            raise ValueError(
+                "Expected the parameters to have one or two dimensions without or with "
+                f"broadcasting, respectively. The parameters have shape {theta_shape}"
+            )
+
+        if theta_shape[-1] != expected_dim:
+            raise ValueError(
+                f"Expected the parameters to have shape ({expected_dim},) or (batch_size, "
+                f"{expected_dim}). The parameters have shape {theta_shape}"
+            )
+
+        super().__init__(theta, wires=wires, do_queue=do_queue, id=id)
+
+    @staticmethod
+    def compute_matrix(theta, num_wires):
+        r"""Representation of the operator as a canonical matrix in the computational basis
+        (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.SpecialUnitary.matrix`
+
+        Args:
+            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`
+            num_wires (int): The number of wires
+
+        Returns:
+            tensor_like: canonical matrix
+
+        .. note::
+
+            Note that this method internally handles a complex-valued tensor of size
+            ``(4**num_wires, 2**num_wires, 2**num_wires)``, which requires at least
+            ``4 ** (2 * num_wires - 13)`` GB of memory (at default precision).
+
+        **Example**
+
+        >>> theta = np.array([0.5, 0.1, -0.3])
+        >>> qml.SpecialUnitary.compute_matrix(theta, num_wires=1)
+        array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
+               [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]])
+        """
+        return special_unitary_matrix(theta, num_wires)
+
+    @staticmethod
+    def compute_decomposition(theta, wires, num_wires):
+        r"""Representation of the operator as a product of other operators (static method).
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        This ``Operation`` is decomposed into the corresponding ``QubitUnitary``.
+
+        .. seealso:: :meth:`~.SpecialUnitary.decomposition`.
+
+        Args:
+            theta (tensor_like): Pauli coordinates of the exponent :math:`A(\theta)`
+            wires (Iterable[Any] or Wires): the wire(s) the operation acts on
+            num_wires (int): The number of wires
+
+        Returns:
+            list[Operator]: decomposition of the operator
+
+        **Example:**
+
+        >>> theta = np.array([0.5, 0.1, -0.3])
+        >>> qml.SpecialUnitary.compute_decomposition(theta, 0, num_wires=1)
+        [QubitUnitary(array([[ 0.83004499-0.28280371j,  0.0942679 +0.47133952j],
+            [-0.0942679 +0.47133952j,  0.83004499+0.28280371j]]), wires=[0])]
+        """
+        return [qml.QubitUnitary(special_unitary_matrix(theta, num_wires), wires=wires)]
```

Comment:
If we avoided using static methods (use `decomposition` method instead) we could do:
```suggestion
        return [qml.QubitUnitary(self.matrix(), wires=self.wires)]
```

Reply:
The current operation abstraction discourages this, I believe. Whenever we want to compute something, the `compute_...` methods are used. I do not know too much about the discussion regarding this, but this was a conscious decision in the operator refactor last year.

Reply:
I'd love to hear why the use of these static methods is encouraged.

For the new operator arithmetic we always use `matrix` and `decomposition` instead.

Reply:
Static methods are dangerous, because it forces any other method used in the pipeline to also be static (loosing all the benefits that a class gives you).

Reply:
I see, that makes a lot of sense.

Reply:
I find it better to *define* the static methods where possible but *use* the instance method. The instance method will be defined if the static method is, but the instance method could be defined without the static method being defined.
Sometimes in the cases of long data lists or long hyperparameter lists, its just a bit more convenient to use the static method instead.

Reply:
Alright! So in this case, where accessing `self.matrix()` is convenient, it would make sense to define `decomposition`, right? :)

Reply:
I went for the `decomposition` implementation. Resolving this.